src/components/NotesMindMapView/NotesMindMapViewSagas.js

```diff
@@ -17,13 +21,67 @@ function* setRootNote() {
 function* requestNoteText({ data }) {
   yield put(changeNoteTextAction(LOADING_NOTE_MESSAGE));
   const noteText = yield call(noteStorage.getNoteContent, data);
-  console.debug('NoteContentController.loadNote(), loaded note content: ', noteText);
   yield put(changeNoteTextAction(noteText));
 }
 
+
+/**
+ * Load child notes for clicked note,
+ * and redraw the network for new notes.
+ */
+function* changeNote({ data: { targetNote, brainVisNetwork } }) {
+  if (!_.isEmpty(targetNote.children)) {
+    renderChildren(targetNote.children, targetNote, brainVisNetwork);
+  } else {
+    const childNote = yield fetchChildNotes(targetNote);
+    renderChildren(childNote, targetNote, brainVisNetwork);
+  }
+
+  saveNote(targetNote);
+
+  if (targetNote.parent) {
+    if (!_.isEmpty(targetNote.parent.name)) {
+      brainVisNetwork.renderParentNote(targetNote);
+    } else {
+      targetNote.parent = yield fetchParentNote(targetNote.id);
+      brainVisNetwork.renderParentNote(targetNote);
+    }
+  }
+}
+
+function* fetchParentNote(noteId) {
+  const fetchingParentNote = spinner.create('loading parent note');
+  const parentNote = yield noteStorage.fetchParentNote(noteId);
+  fetchingParentNote.hide();
+  return parentNote;
+}
+
+function* saveNote(note) {
+  put(changeSelectedNoteAction(note));
+  if (note.isNote) {
+    yield requestNoteText(note);
+  }
+}
+
+function* fetchChildNotes(note) {
+  const fetchingNotesSpinner = spinner.create('loading child notes');
+  fetchingNotesSpinner.show();
+  const childNote = yield noteStorage.fetchChildNotes(note);
+  fetchingNotesSpinner.hide();
+  return childNote;
+}
+
+function renderChildren(children, targetNote, brainVisNetwork) {
```

Comment:
should we have render functions in a separate file? So we can then easily change the rendering engine?